### ISSUE TITLE
Allow chaining files for knarly-long yamls

### ISSUE THREAD

**Issue by ddproxy** [jeffijoe/yenv#4]
env.yaml
 -> Import server.yaml
 -> Import routes.yaml
 -> Import database.yaml

```yaml
server: ~import(server.yaml)
routes: ~import(routes.yaml)
database: ~import(database.yaml)
```

Unfolds to
```yaml
server:
  domain: thing.com
  port: 8080
routes:
  root: '/'
  whatever: '/idk'
database:
  host: host:5432
  user: myuser
```

**Comment by ddproxy** [comment]
@jeffijoe Label: Enhancement

**Comment by jeffijoe** [comment]
You have to remember that the first level of the yaml file is the environment section which maps to whatever `NODE_ENV` is set to. So you're suggesting that imported files do not follow that? Because you'd then need a `database.development.yaml`, and a `database.production.yaml`.

So basically this is what I had in mind:

*database.yaml*

```yaml
development:
  DB_HOST: localhost
  DB_PORT: 1337
```

*server.yaml*

```yaml
development:
  SERVER: localhost
  PORT: 1337

# Remember kids, don't put production variables in your config...
production:
  SERVER: some.site.com
  PORT: 80
```

*env.yaml*

```yaml
~imports: [database.yaml, server.yaml]
development:
  SOME: other setting
```

That way each yaml file get to define sections. The `env.yaml` file could compose the sections of files that were imported.

@ddproxy what do you think?

**Comment by ddproxy** [comment]
That looks great, would allow for fine-tweaking in the env.yaml if necessary to override the imported configs.

**Comment by jeffijoe** [comment]
Implemented in 1.0.
